google_shopping_functions.js

```diff
@@ -1,30 +1,82 @@
 //define getItemsCount function
 //accepts full item data
 //returns the length of the items array
+var data = require("./products.json")
+
 var getItemsCount = function(itemData){
+	for(items in itemData){
+
+	}
 	return itemData.items.length;
 };
 
-//
-//DEFINE FUNCTIONS HERE
-//
+//1. Get Items
+
+var getItems = function(objectData){
+		return objectData.items;
+}
+
+var itemsArray = getItems(data);
 
+//2.  Search Items by Brand
 
+var getItemsByBrand = function(items, brand){
+	var matchedItems = [];
+	for(var i = 0; i < items.length; i++){
+		if(items[i].product.brand === brand){
+			matchedItems.push(items[i]);
+		}
+	}
+	return matchedItems;
+}
 
+//3.  Search items by Author
 
+var getItemsByAuthor = function(items, author){
+	var matchedItems = [];
+	for(var i = 0; i < items.length; i++){
+		if(items[i].product.author.name.toLowerCase().indexOf(author) != -1){
+			matchedItems.push(items[i]);
+		}
+	}
+	return matchedItems;
+}
+
+//4.  Search items by availability
+
+var getItemsByAvailability = function(items){
```

Comment:
To make this function more versatile, you could pass your availability status as an argument, like this:
```
var getItemsByAvailability = function(items, status) {
...
if (items[i].product.inventories[0].availability === status) {
```